src/proxy_app/routing/token_budget.py

```
@@ -0,0 +1,177 @@
+    safety_margin: int = DEFAULT_SAFETY_MARGIN,
+    overrides: Optional[Dict[str, int]] = None,
+) -> int:
+    """Return the effective max_tokens cap for a request's task class.
+
+    ``context_limit`` is the provider's max context window (tokens). The result
+    is always >= FLOOR_TOKEN_BUDGET and <= class_default.
+    """
+    overrides = overrides or env_overrides()
+    base = overrides.get(task_class) or (class_defaults or CLASS_DEFAULTS).get(task_class, 512)
+    mult = _tier_multiplier(tier, tier_multiplier or {})
+    base = max(1, int(round(base * mult)))
```

Ruff (0.16.1)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)